src/main/java/de/eddyson/tapestry/react/services/nashorn/NashornReactRenderEngineImplementation.java

```diff
@@ -0,0 +1,124 @@
+  
+  private ScriptEngine _engine;
+  private SimpleBindings _bindings;
+
+  public NashornReactRenderEngineImplementation(ModuleManager moduleManager, StreamableResourceSource srs,
+      ResourceChangeTracker tracker, AssetSource assetSource) {
+    super();
+    this.moduleManager = moduleManager;
+    this.srs = srs;
+    this.tracker = tracker;
+    this.assetSource = assetSource;
+  }
+
+  @Override
+  public String renderReactComponent(String moduleName, JSONObject parameters) {
+    Stopwatch watch = Stopwatch.createStarted();
+    try {
+      StringBuilder builder = new StringBuilder();
+      builder.append("var reactResult = null;\n");
+      builder.append(
+          "require(['eddyson/react/isomorphic/run', '%1$s'], function(ri) { reactResult = ri('%1$s', %2$s); } );\n");
+      builder.append("reactResult;");
+      String script = String.format(builder.toString(), moduleName, parameters.toCompactString());
+      ScriptEngine engine = getEngine();
+      Bindings bindings = copyGlobalBindings();
+      Object result = engine.eval(script, bindings);
+      if (result == null) {
+        resetEngine();
+        return null;
+      }
+      return result.toString();
+    } catch (Throwable e) {
+      resetEngine();
+      throw new RuntimeException(e);
+    } finally {
+      log.info("Rendering module '" + moduleName + "' took " + watch.toString());
+    }
+  }
+
+  // when an exception occurs while rendering react component discard the engine
+  // require.js set's some internal timeout values for modules that could not be loaded and
+  // throws an exception on next invocation of that module
+  private void resetEngine() {
+    this._engine = null; 
+  }
+
+  // TODO: check if necessary or if fresh bindings would work too...
+  private Bindings copyGlobalBindings() {
+    Bindings bindingsCopy = new SimpleBindings();
+    bindingsCopy.putAll(this._bindings);
+    return bindingsCopy;
+  }
+
+  private ScriptEngine getEngine() throws ScriptException, IOException {
+    if (this._engine == null) {
+      synchronized (this) {
+        if (this._engine == null) {
+          this._engine = buildEngine();
+        }
+      }
+    }
+    return this._engine;
+  }
+
+  private ScriptEngine buildEngine() throws ScriptException, IOException {
+    ScriptEngine nashorn = new ScriptEngineManager().getEngineByName("nashorn");
+    Map<String, Object> data = Maps.newHashMap();
+    this._bindings = new SimpleBindings(data);
+    
+    // "__tapestry" is available in JavaScript code to load AMD modules from Tapestry's ModuleManager
+    data.put("__tapestry", buildNashornModuleLoader(nashorn));
+    
+    // polyfill.js contains some basic polyfills
+    nashorn.eval(read("META-INF/modules/eddyson/react/isomorphic/polyfill.js"), this._bindings);
+    
+    // r.js is the server side implementaiton of require.js for Nashorn/Rhino/Node
+    nashorn.eval(read("META-INF/modules/eddyson/react/isomorphic/r.js"), this._bindings);
```

Comment:
When i tried to get things running i had to fiddle around in r.js and changed some code. But in the end this is not necessary, so i am going to provide r.js via webjar :-)